lib/rules/ban-types.js

```diff
@@ -0,0 +1,99 @@
+    meta: {
+        docs: {
+            description: "Enforces that types will not to be used.",
+            extraDescription: [util.tslintRule("ban-types")],
+            category: "TypeScript",
+            url:
+                "https://github.com/nzakas/eslint-plugin-typescript/blob/master/docs/rules/ban-types.md"
+        },
+        fixable: "code",
+        messages: {
+            bannedTypeMessage:
+                "Don't use '{{name}}' as a type.{{customMessage}}"
+        },
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    types: {
+                        type: "object",
+                        additionalProperties: {
+                            oneOf: [
+                                { type: "null" },
+                                { type: "string" },
+                                {
+                                    type: "object",
+                                    properties: {
+                                        message: { type: "string" },
+                                        fixWith: { type: "string" }
+                                    },
+                                    additionalProperties: false
+                                }
+                            ]
+                        }
+                    }
+                },
+                additionalProperties: false
+            }
+        ]
+    },
+
+    create(context) {
+        const banedTypes = (context.options[0] || {}).types || {};
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        return {
+            "TSTypeReference Identifier"(node) {
+                if (node.parent && node.parent.type !== "TSQualifiedName") {
+                    if (node.name in banedTypes) {
+                        let customMessage = "";
+                        const bannedCfgValue = banedTypes[node.name];
+                        let fixWith = null;
+
+                        if (typeof bannedCfgValue === "string") {
+                            customMessage += ` ${bannedCfgValue}`;
+                        } else if (bannedCfgValue !== null) {
+                            if (bannedCfgValue.message) {
+                                customMessage += ` ${bannedCfgValue.message}`;
+                            }
+                            if (bannedCfgValue.fixWith) {
+                                fixWith = bannedCfgValue.fixWith;
+                            }
+                        }
+
+                        context.report({
+                            node,
+                            messageId: "bannedTypeMessage",
+                            data: {
+                                name: node.name,
+                                customMessage
+                            },
+                            fix(fixer) {
```

Comment:
would it be better to conditionally define the fixer, instead of having the fixer return conditionally?

i.e.
```
{
  fix: fixWith !== null && fixer => fixer.replaceText(node, fixWith)
}
```